addons/project/models/project_task.py

```diff
@@ -301,6 +310,24 @@ def SELF_READABLE_FIELDS(self):
     def SELF_WRITABLE_FIELDS(self):
         return PROJECT_TASK_WRITABLE_FIELDS
 
+    def _inverse_project_id(self):
+        for task in self:
+            if task.project_id:
+                no_display_subtasks = task.child_ids.filtered(lambda t: not t.display_in_project)
```

Comment:
```suggestion
                no_display_subtasks = task.child_ids.filtered(lambda t: not t.display_in_project and t.project_id != task.project_id)
```

In fact, I am wondering if it should not be a compute since the user could: 

- Create a task in project A
- Create a subtask
- change the project of the task to project B

By doing that, we could expect the project set on the sub-task would be Project B as well, but it is not the case since there is no onchange/compute to update the project_id in that use case. 😬